packages/bot-strategy-runner/src/ConfigBuilder.ts

```diff
@@ -0,0 +1,203 @@
+
+  // If this particular bot type is not enabled, return early.
+  if (!config[settingsKey].enableBotType) return [];
+
+  let botTypeWhitelist = globalWhitelist;
+
+  if (config[settingsKey].addressWhitelist)
+    botTypeWhitelist = [...globalWhitelist, ...config[settingsKey].addressWhitelist];
+
+  if (config[settingsKey].addressBlacklist)
+    botTypeWhitelist = botTypeWhitelist.filter(el => !config[settingsKey].addressBlacklist.includes(el));
+
+  const botConfigs: any = [];
+  botTypeWhitelist.forEach((contractAddress: string) => {
+    const addressConfig = mergeConfig(
+      config.commonConfig,
+      config[settingsKey].commonConfig,
+      config[settingsKey].addressConfigOverride ? config[settingsKey].addressConfigOverride[contractAddress] : null
+    );
+    const commonConfig = {
+      botType,
+      syntheticSymbol: syntheticSymbols[contractAddress],
+      botIdentifier: `${syntheticSymbols[contractAddress]} ${botType}`,
+      botNetwork: config.botNetwork,
+      financialContractAddress: contractAddress,
+      errorRetries: addressConfig.errorRetries,
+      errorRetriesTimeout: addressConfig.errorRetriesTimeout,
+      pollingDelay: 0,
+      startingBlock: addressConfig.startingBlock ? addressConfig.startingBlock : process.env.STARTING_BLOCK_NUMBER,
+      endingBlock: addressConfig.endingBlock ? addressConfig.endingBlock : process.env.ENDING_BLOCK_NUMBER
+    };
+    if (botType == "liquidator") {
+      const botConfig: liquidatorConfig = {
+        ...commonConfig,
+        priceFeedConfig: addressConfig.priceFeedConfig,
+        liquidatorConfig: addressConfig.liquidatorConfig,
+        liquidatorOverridePrice: addressConfig.liquidatorOverridePrice
+      };
+      botConfigs.push(botConfig);
+    }
+
+    if (botType == "disputer") {
+      const botConfig: disputerConfig = {
+        ...commonConfig,
+        priceFeedConfig: addressConfig.priceFeedConfig,
+        disputerConfigConfig: addressConfig.disputerConfigConfig,
+        disputerOverridePrice: addressConfig.disputerOverridePrice
+      };
+      botConfigs.push(botConfig);
+    }
+    if (botType == "monitor") {
+      const botConfig: monitorConfig = {
+        ...commonConfig,
+        optimisticOracleAddress: addressConfig.optimisticOracleAddress,
+        monitorConfigConfig: addressConfig.monitorConfigConfig,
+        tokenPriceFeedConfig: addressConfig.tokenPriceFeedConfig,
+        medianizerPriceFeedConfig: addressConfig.medianizerPriceFeedConfig,
+        denominatorPriceFeedConfig: addressConfig.denominatorPriceFeedConfig
+      };
+      botConfigs.push(botConfig);
+    }
+  });
+  return botConfigs;
+}
+
+// Takes in two objects of any depth and merges them. In the case of a collision in keys the second object will take priority.
+export function mergeConfig(...args: any) {
+  const target: any = {}; // create a new object
+
+  // deep merge the object into the target object
+  const merger = (obj: any) => {
```

Comment:
nit: why not call this method deepMerge?

Reply:
I've removed this method and am using lodash after Davids comment :)